components/ContestFilter.tsx

```diff
@@ -0,0 +1,70 @@
+import React from 'react'
+import SearchInput from './SearchInput'
+
+function ContestFilter(props) {
+  const { open, setOpen, statusFilter, setStatusFilter } = props
+
+  const handleStatusFilter = (status: string) => {
+    setStatusFilter(status)
+  }
+
+  const selectedFilterCSS = (filter) => {
+    if (filter === statusFilter) {
+      return 'text-orange-500 bg-orange-700'
+    }
+    return 'text-gray-700 bg-gray-700'
+  }
+
+  return (
+    <>
+      <div className='flex-row'>
+        <button
+          className='text-gray-700 bg-gray-500 text-white px-3 py-2 rounded-2xl font-medium cursor-pointer mr-2'
+          onClick={() => setOpen(!open)}
+        >
+          Filters
+        </button>
+
+        <SearchInput />
+      </div>
+      {open && (
+        <div>
+          <button
+            className={`${selectedFilterCSS(
+              'All'
+            )} text-white px-3 py-2 rounded-2xl font-medium cursor-pointer mr-2 mt-2`}
+            onClick={() => handleStatusFilter('All')}
+          >
+            All
+          </button>
```

Comment:
the filter buttons can be created with a for loop that iterates over a list of the possible filters ["All", "Open", "Voting", "Closed"]